src/net_processing.cpp

```diff
@@ -988,8 +988,11 @@ class PeerManagerImpl final : public PeerManager
     /** Remove this block from our tracked requested blocks. Called if:
      *  - the block has been recieved from a peer
      *  - the request for the block has timed out
+     * If "from_peer" is specified, then only remove the block if it is in
```

Comment:
27608: consider to backport as well 29412 (regretion tests for 27608) and put them in one PR

Reply:
the agent failed to backport 29412 - https://github.com/DashCoreAutoGuix/dash/pull/712

If you'd like to provide a commit that backports it, I can add it. Otherwise, I'd like to either keep it, or drop it and merge this PR and continue :D